File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Mutation_CreateOrder_Arguments.java
package com.example.grts;

import graphql.schema.GraphQLInputObjectType;
import viaduct.java.api.globalid.GlobalID;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.List;
import java.util.Map;
import viaduct.java.api.types.Arguments;
import viaduct.java.api.internal.InputBase;
import viaduct.java.api.internal.InternalContext;

/** Generated arguments class for resolver field. */
public class Mutation_CreateOrder_Arguments extends InputBase implements Arguments {

    public Mutation_CreateOrder_Arguments(InternalContext context, Map<String, Object> data, GraphQLInputObjectType graphQLInputObjectType) {
        super(context, data, graphQLInputObjectType);
    }

        public CreateOrderInput getInput() {
            return getInput("input", CreateOrderInput::new);
        }

}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Mutation_CreateOrder_Arguments.kt
@file:Suppress("warnings")

package viaduct.api.grts

import graphql.schema.GraphQLInputObjectType
import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InputTypeFactory
import viaduct.api.internal.InputValueBuilder
import viaduct.api.internal.InternalContext
import viaduct.api.internal.internal
import viaduct.api.internal.InputLikeBase
import viaduct.api.types.Input

@OptIn(InternalApi::class)
class Mutation_CreateOrder_Arguments internal constructor(
    override val context: InternalContext,
    override val inputData: Map<String, Any?>,
    override val graphQLInputObjectType: GraphQLInputObjectType,
): InputLikeBase(), viaduct.api.types.Arguments {
    init {
       TODO()
    }

    val input: viaduct.api.grts.CreateOrderInput get() = TODO()


    fun toBuilder() = Builder(context, graphQLInputObjectType, this.inputData.toMutableMap())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): Mutation_CreateOrder_Arguments =
            Builder(context).apply(block).build()
    }

    class Builder internal constructor(
        override val context: InternalContext,
        override val graphQLInputObjectType: GraphQLInputObjectType,
        override val inputData: MutableMap<String, Any?> = TODO()
    ) : InputLikeBase.Builder(), InputValueBuilder<Mutation_CreateOrder_Arguments> {

        constructor(context: ExecutionContext): this(
            context.internal,
            InputTypeFactory.argumentsInputType("Mutation_CreateOrder_Arguments", "Mutation", "createOrder", context.internal.schema),
            mutableMapOf()
        )

        init {
            TODO()
        }

                    fun input(value: viaduct.api.grts.CreateOrderInput): Builder = TODO()


        final override fun build(): Mutation_CreateOrder_Arguments = TODO()
    }

}